sky/shell/platform/ios/platform_view_ios.mm

```diff
@@ -299,9 +380,15 @@ bool UpdateStorageSizeIfNecessary() {
 void PlatformViewIOS::RunFromSource(const std::string& main,
                                     const std::string& packages,
                                     const std::string& assets_directory) {
-  // TODO(johnmccutchan): Call to the iOS UI thread so that services work
-  // properly like we do in PlatformViewAndroid.
-  engine().RunFromSource(main, packages, assets_directory);
+  auto latch = new ftl::ManualResetWaitableEvent();
+
+  dispatch_async(dispatch_get_main_queue(), ^{
+    SetupAndLoadFromSource(main, packages, assets_directory);
+    latch->Signal();
+  });
+
+  latch->Wait();
+  delete latch;
```

Comment:
Why not just allocate the latch on the stack?

I'm losing track of which threads are allowed to wait on which other threads.  We should probably document that somewhere to convince ourselves that we don't have deadlocks.